### PR TITLE
Adding another layer on to the foundation

### PR DESCRIPTION
**Pull request by realstealthninja**
### Changes

- Added base bot 
- made improvements to the main file for user accessibility
- linted code

### CODE REVIEW

**Review by Nipa-Code — APPROVED** [review]
lgtm, good job!

File: main.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import os
+from src import Glykon
+
+if __name__ == "__main__":
+
```
Comment (CaedenPH) [inline main.py:5]:
```suggestion
```

File: main.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    if ".env" not in os.listdir():
+
```
Comment (CaedenPH) [inline main.py:7]:
```suggestion
```

File: main.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        print("⇒ Running configurations")
+
+        token = input("🔴 paste your bot's token 🔴:")
```
Comment (CaedenPH) [inline main.py:10]: I would prefer an error instead, letting the user know that the config is invalid

File: main.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import os
+from src import Glykon
+
+if __name__ == "__main__":
+
+    if ".env" not in os.listdir():
+
+        print("⇒ Running configurations")
```
Comment (CaedenPH) [inline main.py:8]: Instead of printing, you can use the `logging` module

File: main.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+        token = input("🔴 paste your bot's token 🔴:")
+        with open(".env", "x") as f:
+            f.write(f"TOKEN='{token}'")
```
Comment (CaedenPH) [inline main.py:12]: I'm not a fan of writing a token into a `.env`

File: src/bot.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
+"""Main bot file"""
```
Comment (CaedenPH) [inline src/bot.py:1]:
```suggestion
```

File: src/bot.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+from dotenv import load_dotenv
+
+load_dotenv(dotenv_path="./.env")
+TOKEN = os.getenv("TOKEN")
+
+
+class Glykon(Bot):
+    """Bots class"""
+    def __init__(self):
```
Comment (CaedenPH) [inline src/bot.py:14]:
```suggestion

    def __init__(self):
```

File: src/bot.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+            ⤷ Users 👥: {len(self.users)}
+            ⤷ Cogs ⚙️: {len(self.cogs)}
+            ───── ⋆⋅☆⋅⋆ ─────
+                🟢 Ready 🟢
+            """
+        )
+
+    def run(self) -> None:
+        self.load_cogs()
+        super().run(TOKEN, reconnect=True)
```
Comment (CaedenPH) [inline src/bot.py:45]:
```suggestion
        super().run(TOKEN, reconnect=True)
        
```

File: src/__init__.py [hunk 9]
```diff
@@ -0,0 +1 @@
+from .bot import Glykon
```
Comment (CaedenPH) [inline src/__init__.py:1]: You should include `__version__` and other special attributes here

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Instead of printing to the terminal, you should instead use the python logging module;
    https://docs.python.org/3/library/logging.html

In addition, you should install the `pre-commit` hooks when making pull requests, which automatically formats the code and adds new lines to the end of each file.

**Comment by Nipa-Code** [comment]
> Instead of printing to the terminal, you should instead use the python logging module; https://docs.python.org/3/library/logging.html

I would prefer using loguru over python standard logging module. IMO easier and better in general for normal logging (as I have done only normal logging stuff)

**Comment by CaedenPH** [comment]
> > Instead of printing to the terminal, you should instead use the python logging module; https://docs.python.org/3/library/logging.html
> 
> 
> 
> I would prefer using loguru over python standard logging module. IMO easier and better in general for normal logging (as I have done only normal logging stuff)
> 
> 
> 
> 

Sounds fine

File: .pre-commit-config.yaml [hunk 10]
```diff
@@ -29,4 +29,5 @@ repos:
     rev: 21.12b0
     hooks:
       - id: black
-        name: Running black in all files.
\ No newline at end of file
+        name: Running black in all files.
+        additional_dependencies: ["click==8.0.4"]
```
Comment (realstealthninja) [inline .pre-commit-config.yaml:33]: Pinning click to 8.0.4 due to https://github.com/psf/black/issues/2964

File: src/src.md [hunk 11]
```diff
@@ -0,0 +1,3 @@
+### SRC
```
Comment (CaedenPH) [inline src/src.md:1]: This file is completely useless, you use python documentstion to explain code structure, not md files

File: src/bot.py [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+from disnake import Intents
+from disnake.ext.commands import Bot
+from dotenv import load_dotenv
+
+load_dotenv(dotenv_path="./.env")
+TOKEN = os.getenv("TOKEN")
+
+# Logger config
+logger = logging.getLogger("disnake")
```
Comment (CaedenPH) [inline src/bot.py:12]:
```suggestion
logger = logging.getLogger(__name__)
```

File: src/__init__.py [hunk 13]
```diff
@@ -0,0 +1,5 @@
+__title__ = "glykon"
```
Comment (CaedenPH) [inline src/__init__.py:1]:
```suggestion
__name__ = "glykon"
```

File: src/bot.py [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+        )
+
+    def load_cogs(self) -> None:
+        """Load cogs from the src/cogs directory"""
+        for file in os.listdir("./src/cogs"):
+            if file.startswith("_"):
+                continue
+            if not file.endswith(".py"):
+                self.load_extension(f"src.cogs.{file}")
+            self.load_extension(f"src.cogs.{file[:-3]}")
```
Comment (CaedenPH) [inline src/bot.py:44]: I believe there is a `load_extensions` method in `disnake.ext.commands.Bot` now

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Please commit the suggestions that I made

File: src/bot.py (hunk 14 above)
Comment (realstealthninja) [reply src/bot.py:44]: does load_extensions work with cog as folder?

**Review by CaedenPH — APPROVED** [review]

**Pull request merged**